### PR TITLE
Separation of Lexer interface and implementation

### PR DESCRIPTION
**Pull request by Renelvon**
I splitted class LlamaLexer into classes Lexer (interface) and _LexerBuilder (implementation). It is now possible to construct a complete lexer for llama with one call to Lexer (see main.py).

Issues:
1) The lexer returned from _LexerBuilder is somewhat incomplete; it doesn't conform to the PLY interface.
2) The Lexer class is tightly coupled with the _LexerBuilder class.

Both of these issues can be fixed by making _LexerBuilder produce lexers conforming to the PLY interface. This is not hard; just add the appropriate "@property" decorators to the _LexerBuilder class. However, since Lexer class does that already, there will be some inevitable code duplicaton.

### CODE REVIEW

File: lexer.py [hunk 1]
```diff
@@ -164,39 +123,14 @@ def lineno(self, value):
     # Levels of nested comment blocks still open
     level = 0
 
-    def __init__(self, debug=False):
+    def __init__(self, verbose=False):
```
Comment (dionyziz) [inline lexer.py:126]: I like this name change, it makes things less confusing than before when we had multiple debug variables that meant different things.

File: lexer.py [hunk 2]
```diff
@@ -246,27 +180,40 @@ def input(self, lexdata):
...
-        newLexer = LlamaLexer(debug=self.debug)
-        newLexer.lexer = self.lexer.clone()
+        new_lexer = _LexerBuilder(verbose=self.verbose)
+        new_lexer.lexer = self.lexer.clone()
 
-        newLexer.bol = self.bol
-        newLexer.level = self.level
+        new_lexer.bol = self.bol
+        new_lexer.level = self.level
         return newLexer
```
Comment (dionyziz) [inline lexer.py]: undefined variable?

File: lexer.py [hunk 3]
```diff
@@ -246,27 +180,40 @@ def input(self, lexdata):
...
+    def warning_out(self, message, lineno=None, lexpos=None):
+        """Signal lexing warning."""
+        if lineno is not None:
+            if lexpos is not None:
+                s = "%s: %d:%d warning: %s" % (lineno, lexpos, message)
+            else:
+                s = "%s: %d: warning: %s" % (lineno, message)
+        else:
+            s = "%s: warning: %s" % (message)
+        err.push_warning(lineno or 0, s)
```
Comment (dionyziz) [inline lexer.py:216]: DRY; recommend merging `error_out` and `warning_out` into one. Perhaps on a separate PR?

File: lexer.py (hunk 2 above)
Comment (Renelvon) [inline lexer.py]: Right. Maybe I should code some unit tests.

File: lexer.py (hunk 3 above)
Comment (Renelvon) [inline lexer.py:216]: I intend to fix that when I re-implement the error module over Python's logging module.

**Pull request merged**